runtime/src/main/java/org/corfudb/runtime/view/ManagementView.java

```diff
@@ -155,4 +154,54 @@ public void addNode(@Nonnull String endpointToAdd, int retry,
 
         throw new WorkflowResultUnknownException();
     }
+
+    /**
+     * Heal an unresponsive node.
+     *
+     * @param endpointToHeal Endpoint of the new node to be healed in the cluster.
+     * @param retry          the number of times to retry a workflow if it fails
+     * @param timeout        total time to wait before the workflow times out
+     * @param pollPeriod     the poll interval to check whether a workflow completed or not
+     * @throws WorkflowResultUnknownException when the side affect of the operation
+     *                                        can't be determined
+     */
+    public void healNode(@Nonnull String endpointToHeal, int retry, @Nonnull Duration timeout,
+                         @Nonnull Duration pollPeriod) {
+
+        for (int retryAttempt = 0; retryAttempt < retry; retryAttempt++) {
+            try {
+                runtime.invalidateLayout();
+                Layout layout = runtime.getLayoutView().getLayout();
+                // Select the current tail node and send an add node request to the orchestrator
+                List<String> logServers = layout.getSegments().get(0).getStripes().get(0)
+                        .getLogServers();
+                String orchestratorEndpoint = logServers.get(logServers.size() - 1);
```

Comment:
why do we get orchestrator endpoint from logServers. Also why do not have a util function for this ? I think this boiler plate should be moved to a util and should be used the same way everywhere.

Reply:
We get the orchestrator endpoint from the logServers tail to minimize the stateTransfer cost (We reduce the reading cost). This applies to Add and Heal workflows.
Whereas for remove node, the endpoint has to be computed separately to avoid choosing the node to be removed as the orchestrator. I'll extract the boiler plate.